### PR TITLE
Update MongoDB examples according to MongoDB Ruby Driver v2.0.0 new APIs.

### PR DESCRIPTION
**Pull request by thyrlian**

### CODE REVIEW

File: databases/mongo.md [hunk 1]
```diff
@@ -26,17 +26,14 @@ require 'sinatra'
...
 end
 ```
 
-With a connection "in hand" you can connect to any database and collection in
-your Mongo instance.
+After creating a client to a MongoDB server, you can use the driver to
+communicate with the database.
 
 ```ruby
 get '/collections/?' do
```
Comment (kgrz) [inline databases/mongo.md]: I think this `collections` handler also requires some changes: https://www.dropbox.com/s/5tjdxmkdqrw0k9i/Screenshot%202015-06-29%2020.30.09.png?dl=0

File: databases/mongo.md [hunk 2]
```diff
@@ -26,17 +26,14 @@ require 'sinatra'
...
 require 'json/ext' # required for .to_json
 
-include Mongo
-
 configure do
-  conn = MongoClient.new("localhost", 27017)
-  set :mongo_connection, conn
-  set :mongo_db, conn.db('test')
+  db = Mongo::Client.new([ '127.0.0.1:27017' ], :database => 'test')  
+  set :mongo_db, db
```
Comment (kgrz) [inline databases/mongo.md]: I feel it would be worth changing this to `set :mongo_db, db['test']` and get rid of all the `mongo_db['test']` calls.

**Comment by kgrz** [comment]
LGTM except for those two comments. Thanks for this.

**Comment by thyrlian** [comment]
Thanks @kgrz for your response.

1. I just checked again the current API, `collection_names` is still valid, just as your screenshots, which returns collection names as strings.  And the other `collections` would return an array of collection objects.  Do you mean that I should change to the latter one?  Thanks for more explanation.

2. Yes, changing it to `set :mongo_db, db['test']` makes sense if the example code only deals with test database.  Sorry I didn't do that before because I just want to make the minimum change.

I'll wait for your reply and do corresponding change then submit a pull request again.  Thanks a lot.

**Comment by thyrlian** [comment]
@kgrz I got your 1st point after debugging by myself.  Yes, you're right, now user has to call `database` before calling `collection_names`.  Thanks for your check.  I'll add these two changes to send another pull request.

**Comment by thyrlian** [comment]
@kgrz I've made corresponding changes according to your comments (and tested).  Please check and merge.  Thanks.

**Comment by thyrlian** [comment]
Hi @kgrz could you please review and merge it?  Thanks.

**Comment by kgrz** [comment]
@thyrlian Thank you! Sorry I kept this waiting.

**Comment by thyrlian** [comment]
@kgrz Thanks for merging :+1:

**Pull request merged**